src/main.rs

```diff
@@ -31,28 +27,26 @@ async fn main() -> Result<(), LemmyError> {
   env_logger::init();
   let settings = Settings::get();
 
-  // Set up the r2d2 connection pool
+  // Set up the deadpool connection pool
   let db_url = match get_database_url_from_env() {
     Ok(url) => url,
     Err(_) => settings.get_database_url(),
   };
-  let manager = ConnectionManager::<PgConnection>::new(&db_url);
-  let pool = Pool::builder()
-    .max_size(settings.database.pool_size)
-    .build(manager)
-    .unwrap_or_else(|_| panic!("Error connecting to {}", db_url));
+
+  let manager = Manager::new(db_url);
+  let pool = Pool::new(manager, settings.database.pool_size);
+  let conn = &pool.get().await?;
 
   // Run the migrations from code
-  blocking(&pool, move |conn| {
-    embedded_migrations::run(conn)?;
-    run_advanced_migrations(conn)?;
-    Ok(()) as Result<(), LemmyError>
-  })
-  .await??;
+  embedded_migrations::run(conn)?;
+  run_advanced_migrations(conn)?;
 
-  let pool2 = pool.clone();
+  // TODO can't move the pool into clokwerk, it doesn't yet support async
+  let c1 = pool.get().await?;
+  let c2 = pool.get().await?;
+  let c3 = pool.get().await?;
   thread::spawn(move || {
-    scheduled_tasks::setup(pool2);
+    scheduled_tasks::setup(c1, c2, c3);
   });
```

Comment:
clokwerk doesn't have async supported / documented yet. This is annoying but it works fine. I tried for like an hour to move the pool into this thread, but I couldn't get it working with tokio. It would often compile fine, but then lock the threads on running.